src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2056,28 +2061,29 @@ public Result(DecoratedKey key, TableMetadata metadata, int z, String writerId,
                     {
                         writerId = ByteBufferUtil.string(c.value());
                     }
-                    catch (Exception e)
+                    catch (CharacterCodingException e)
                     {
+                        logger.error("write Id could not be read");
                     }
+                    TableMetadata tableMetadata = ri.metadata();
```

Comment:
can this line be inserted before 2044? I think the metadata is not changed in the loop.

Reply:
I think it can I will change it in the next commit